### PR TITLE
[Review][Speech] Add Speech Framework from Xcode 8 Beta 1

### PR DESCRIPTION
**Pull request by dalexsoto**
FIXME: SFSpeechRecordingRecognitionRequest is not in the public api
filled radar://26799291 https://trello.com/c/s6s6YKua

### CODE REVIEW

**Comment by xamarin-release-manager** [comment]
Build success

File: src/callkit.cs [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+		Unknown = 0,
+		Disabled = 1,
+		Enabled = 2
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.MacOSX, 10, 12)]
+	[Native]
+	public enum CXErrorCode : nint {
+		UnknownError = 0
```
Comment (migueldeicaza) [inline src/callkit.cs]: Let us rename to "Unknown", as the "Error" is redundant there.

File: src/callkit.cs [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+		void GetEnabledStatusForExtension (string identifier, Action<CXCallDirectoryEnabledStatus, NSError> completion);
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.MacOSX, 10, 12)]
+	[BaseType (typeof (NSObject))]
+	interface CXCallDirectoryProvider {
+
+		[Export ("beginRequestWithExtensionContext:")]
+		void BeginRequestWithExtensionContext (CXCallDirectoryExtensionContext context);
```
Comment (migueldeicaza) [inline src/callkit.cs]: Drop "WithExtensionContext" from the method name.

File: src/callkit.cs [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+
+		[Export ("providerDidBegin:")]
+		void ProviderDidBegin (CXProvider provider);
+
+		[Export ("provider:executeTransaction:")]
+		bool ExecuteTransaction (CXProvider provider, CXTransaction transaction);
+
+		[Export ("provider:performStartCallAction:")]
+		void StartCallAction (CXProvider provider, CXStartCallAction action);
+
```
Comment (migueldeicaza) [inline src/callkit.cs]: I think we should have "Perform" in the prefix here and the few methods below.

File: src/callkit.cs (hunk 1 above)
Comment (dalexsoto) [inline src/callkit.cs]: Fixed in https://github.com/xamarin/xamarin-macios/pull/189

File: src/callkit.cs (hunk 2 above)
Comment (dalexsoto) [inline src/callkit.cs]: Fixed in https://github.com/xamarin/xamarin-macios/pull/189

File: src/callkit.cs (hunk 3 above)
Comment (dalexsoto) [inline src/callkit.cs]: https://github.com/xamarin/xamarin-macios/pull/189

**Comment by dalexsoto** [comment]
New PR

**Pull request closed without merging**